input/posts/2020-09-02-migrating-wcf-to-grpc.md

```diff
@@ -0,0 +1,507 @@
+Title: Migrating WCF to gRPC - The protobuf-net way
+Published: 2020-09-02
+Tags: 
+- .NET
+- gRPC
+- Protocol Buffers
+- WCF
+---
+
+In my [previous blog post](2020-07-08-grpc-reflection-in-net), I mentioned that it's straightforward to migrate your WCF services to gRPC using [protobuf-net.Grpc](https://github.com/protobuf-net/protobuf-net.Grpc/). In this blog post, we are going to look at how easy it actually is. Microsoft's [official guide for migrating WCF services to gRPC](https://docs.microsoft.com/en-us/dotnet/architecture/grpc-for-wcf-developers/migrate-wcf-to-grpc) only mentions the Google.Protobuf approach, which can be time consuming if you have a lot of data contracts that need to be migrated to `.proto` format. However, by using [protobuf-net.Grpc](https://github.com/protobuf-net/protobuf-net.Grpc/) we are be able to reuse the old WCF data contracts and service contracts with minimal code changes.
```

Comment:
Last sentence: "we are able to reuse..." (there's a "be" in there that should be removed)